highlight/highlight
# Celery: tracing integration

## Summary
- Requires Python version 3.8 or newer.
- Support tracing auto-instrumention for the celery library.
- Optimizations for requests auto-instrumentation tracing.
- Removes `tracing_origins` configuration for requests library and passes by default.

## How did you test this change?
1. Start the fastapi e2e app (consumer and worker)
2. Hit the endpoint to start a celery job: `curl -H "x-highlight-request: cs5/cr5" http://localhost:8000/celery`
3. Confirm the tracing works on the Highlight UI
- [ ] Span added for publisher
- [ ] Span added for consumer
- [ ] Both spans are linked to same trace and session
4) Start the flask app
5) Hit the endpoint to make an external request: `curl -H "x-highlight-request: r39/r39" http://localhost:8085/external`
6) Confirm the tracing works on the Highlight UI
- [ ] Span created for request
- [ ] Span are linked to correct trace and session

https://www.loom.com/share/ae9fefe78caf4c67967a3b64d5a969a0?sid=4323f805-3dbf-49c2-8194-74726087a46d

## Are there any deployment considerations?
N/A

## Does this work require review from our design team?
N/A
